### PR TITLE
[ISSUE #5096] Add unit tests to package com.alibaba.nacos.naming.healthcheck in nacos 2.0

### PR DESCRIPTION
**Pull request by li-xiao-shuang**
a part of fixing ISSUE #5096 
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by KomachiSion** [comment]
Please sign your CLA first

File: naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/HealthCheckTaskV2Test.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    private ConfigurableApplicationContext context;
+    
+    @Mock
+    private SwitchDomain switchDomain;
+    
+    @Mock
+    private Service service;
+    
+    @Before
+    public void initHealthCheckTaskV2() {
```
Comment (KomachiSion) [inline naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/HealthCheckTaskV2Test.java:57]: Before method name should be setUp

File: naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/HealthCheckTaskV2Test.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        
+        healthCheckTaskV2.run();
+        healthCheckTaskV2.passIntercept();
+        healthCheckTaskV2.doHealthCheck();
+        
+        verify(ipPortBasedClient, times(3)).getAllPublishedService();
+        verify(switchDomain, times(3)).isHealthCheckEnabled(service.getGroupedServiceName());
+    }
+    
+    private List<Service> returnService() {
```
Comment (KomachiSion) [inline naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/HealthCheckTaskV2Test.java:82]: Can replace with Collections.singleList

File: naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HealthCheckCommonV2Test.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    @Mock
+    private IpPortBasedClient ipPortBasedClient;
+    
+    @Mock
+    private HealthCheckInstancePublishInfo healthCheckInstancePublishInfo;
+    
+    private HealthCheckCommonV2 healthCheckCommonV2;
+    
+    @Before
+    public void init() {
```
Comment (KomachiSion) [inline naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HealthCheckCommonV2Test.java:57]: Before method name should be setUp

File: naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HealthCheckCommonV2Test.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        healthCheckCommonV2 = new HealthCheckCommonV2();
+        when(healthCheckTaskV2.getClient()).thenReturn(ipPortBasedClient);
+        when(ipPortBasedClient.getInstancePublishInfo(service)).thenReturn(healthCheckInstancePublishInfo);
+        when(healthCheckInstancePublishInfo.getFailCount()).thenReturn(new AtomicInteger());
+    }
+    
+    @Test
+    public void reEvaluateCheckRT() {
```
Comment (KomachiSion) [inline naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HealthCheckCommonV2Test.java:65]: test method should start with `test`

File: naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HealthCheckProcessorV2DelegateTest.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    @Mock
+    private Service service;
+    
+    @Mock
+    private ClusterMetadata clusterMetadata;
+    
+    private HealthCheckProcessorV2Delegate healthCheckProcessorV2Delegate;
+    
+    @Before
+    public void initBean() {
```
Comment (KomachiSion) [inline naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HealthCheckProcessorV2DelegateTest.java:58]: Before method name should be setUp

File: naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HealthCheckProcessorV2DelegateTest.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        Class<HealthCheckProcessorV2Delegate> healthCheckProcessorV2DelegateClass = HealthCheckProcessorV2Delegate.class;
+        try {
+            Field field = healthCheckProcessorV2DelegateClass.getDeclaredField("healthCheckProcessorMap");
+            field.setAccessible(true);
+            Map<String, HealthCheckProcessorV2> map = (Map<String, HealthCheckProcessorV2>) field
+                    .get(healthCheckProcessorV2Delegate);
+            HealthCheckProcessorV2 healthCheckProcessorV2 = map.get(HealthCheckType.TCP.name());
+            Assert.assertNotNull(healthCheckProcessorV2);
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            e.printStackTrace();
```
Comment (KomachiSion) [inline naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HealthCheckProcessorV2DelegateTest.java:78]: Why ignore exception? if throw exception is expected behavior, should add it in @Test(expected=xxx)

File: naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HttpHealthCheckProcessorTest.java [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+    @Mock
+    private RestResult restResult;
+    
+    @Mock
+    private ConnectException connectException;
+    
+    private HttpHealthCheckProcessor httpHealthCheckProcessor;
+    
+    @Before
+    public void initBean() {
```
Comment (KomachiSion) [inline naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HttpHealthCheckProcessorTest.java:79]: Before method name should be setUp

File: naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HttpHealthCheckProcessorTest.java [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+        try {
+            Constructor<?> constructor = aClass
+                    .getConstructor(HttpHealthCheckProcessor.class, HealthCheckInstancePublishInfo.class,
+                            HealthCheckTaskV2.class, Service.class);
+            Object objects = constructor
+                    .newInstance(httpHealthCheckProcessor, healthCheckInstancePublishInfo, healthCheckTaskV2, service);
+            
+            Assert.assertNotNull(objects);
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (KomachiSion) [inline naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HttpHealthCheckProcessorTest.java:113]: Same problem as last exception handle

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HttpHealthCheckProcessorTest.java (hunk 7 above)
Comment (li-xiao-shuang) [reply naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HttpHealthCheckProcessorTest.java]: > Before method name should be setUp

fix

File: naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HealthCheckCommonV2Test.java (hunk 3 above)
Comment (li-xiao-shuang) [reply naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HealthCheckCommonV2Test.java]: > Before method name should be setUp

fix

File: naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/HealthCheckTaskV2Test.java (hunk 2 above)
Comment (li-xiao-shuang) [reply naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/HealthCheckTaskV2Test.java:82]: > Can replace with Collections.singleList

fix

File: naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HealthCheckCommonV2Test.java (hunk 4 above)
Comment (li-xiao-shuang) [reply naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HealthCheckCommonV2Test.java]: > test method should start with `test`

fix

File: naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HealthCheckProcessorV2DelegateTest.java (hunk 5 above)
Comment (li-xiao-shuang) [reply naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HealthCheckProcessorV2DelegateTest.java]: > Before method name should be setUp

fix

File: naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HttpHealthCheckProcessorTest.java (hunk 8 above)
Comment (li-xiao-shuang) [reply naming/src/test/java/com/alibaba/nacos/naming/healthcheck/v2/processor/HttpHealthCheckProcessorTest.java]: > Same problem as last exception handle

fix

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**